akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -116,10 +99,30 @@ private[akka] class DaemonMsgCreateSerializer(val system: ExtendedActorSystem) e
 
     def props = {
       import scala.collection.JavaConverters._
-      val clazz = system.dynamicAccess.getClassFor[AnyRef](proto.getProps.getClazz).get
-      val args: Vector[AnyRef] = (proto.getProps.getArgsList.asScala zip proto.getProps.getClassesList.asScala)
-        .map(deserialize)(collection.breakOut)
-      Props(deploy(proto.getProps.getDeploy), clazz, args)
+      val protoProps = proto.getProps
+      val actorClass = system.dynamicAccess.getClassFor[AnyRef](protoProps.getClazz).get
+      val args: Vector[AnyRef] =
+        if (protoProps.getSerializerIdsCount > 0) {
+          // message from a 2.5+ node, which (may) includes string manifest and serializer id
+          for {
+            idx ← (0 until protoProps.getSerializerIdsCount).toVector
+          } yield {
+            val manifest = protoProps.getClasses(idx)
+            // we have info per position if a string manifest serializer was used or not
+            if (protoProps.getHasManifest(idx)) {
+              serialization.deserializeByteBuffer(
+                protoProps.getArgs(idx).asReadOnlyByteBuffer(),
+                protoProps.getSerializerIds(idx), manifest)
+            } else {
+              oldDeserialize(protoProps.getArgs(idx), protoProps.getClasses(idx))
```

Comment:
here it would be better to use the serializerId, i.e.

```
serialization.deserializeByteBuffer(
  protoProps.getArgs(idx).asReadOnlyByteBuffer(),
  protoProps.getSerializerIds(idx),
  manifest = "")
```

I think the whole thing can be rewritten as:
```
val manifest = if (protoProps.getHasManifest(idx)) protoProps.getClasses(idx) else ""
serialization.deserializeByteBuffer(
  protoProps.getArgs(idx).asReadOnlyByteBuffer(),
  protoProps.getSerializerIds(idx), 
  manifest)
```

given that the `hasManifest` is set as in my other comment in `toBinary`